cis.py

```diff
@@ -16,191 +15,249 @@
+                             'derivatives.')
+    parser.add_argument('-w', '--workdir',
+                        required=False,
+                        dest='work_dir',
                         default=None,
                         help='Path to a working directory. Defaults to work '
                              'subfolder in dset_dir.')
-    parser.add_argument('--config', required=True, dest='config',
+    parser.add_argument('--config',
+                        required=True,
+                        dest='config',
                         help='Path to the config json file.')
-    parser.add_argument('--protocol_check', required=False, action='store_true',
-                        help='Will perform a protocol check to determine if the correct number of scans and TRs are present.')
-    parser.add_argument('--autocheck', required=False, action='store_true',
-                        help='Will automatically download all scans from XNAT that are not currently in the project folder.')
-    parser.add_argument('--xnat_experiment', required=False, dest='xnatexp',
+    parser.add_argument('--protocol_check',
+                        required=False,
+                        action='store_true',
+                        help='Will perform a protocol check to determine if '
+                             'the correct number of scans and TRs are present.')
+    parser.add_argument('--autocheck',
+                        required=False,
+                        action='store_true',
+                        help='Will automatically download all scans from XNAT '
+                             'that are not currently in the project folder.')
+    parser.add_argument('--xnat_experiment',
+                        required=False,
+                        dest='xnatexp',
                         default=None,
-                        help='XNAT Experiment ID (i.e., XNAT_E*) for single session download.')
-    parser.add_argument('--n_procs', required=False, dest='n_procs',
+                        help='XNAT Experiment ID (i.e., XNAT_E*) for single '
+                             'session download.')
+    parser.add_argument('--n_procs',
+                        required=False,
+                        dest='n_procs',
                         help='Number of processes with which to run MRIQC.',
-                        default=1, type=int)
+                        default=1,
+                        type=int)
     return parser
 
 
-def main(argv=None):
-    args = get_parser().parse_args(argv)
-
+def main(bids_dir, config, work_dir=None, protocol_check=False, autocheck=False,
+         xnatexp=None, n_procs=1):
     CIS_DIR = '/scratch/cis_dataqc/'
 
     # Check inputs
-    if args.work_dir is None:
-        args.work_dir = CIS_DIR
+    if work_dir is None:
+        work_dir = CIS_DIR
 
-    proj_dir = os.path.dirname(args.bids_dir)
+    proj_dir = os.path.dirname(bids_dir)
     if not op.isdir(proj_dir):
```

Comment:
As we've done for `cis-bidsify` I think we should rework all the `os.path` instances to `pathlib`

Reply:
That should wait until the next PR. I'll open an issue.

Reply:
Actually scratch that. cis-processing needs to work with minimal dependencies, as well as Python 2.7 (still currently the default on the HPC).

Reply:
I feel like going with the 2.7.5 on the HPC is really limiting, is it unreasonable for us to require an environment with Python 3.6 or greater? We can even heavily link to information on setting up basic environments on the HPC and using them in submission scripts.

Reply:
The purpose of cis-processing is to act as the glue between workflows with real dependencies (packaged in Singularity images), with absolutely minimal dependencies. cis-processing shouldn't be doing any processing. It just copies files, builds and runs a series of Singularity calls, and copies things back. Any real processing should go into `cis-bidsify` or another Singularity-based workflow.